pkg/controller/tiers/tiers_controller.go

```diff
@@ -170,38 +168,42 @@ func (r *ReconcileTiers) prepareTiersConfig(ctx context.Context, reqLogger logr.
 		DNSEgressCIDRs: tiers.DNSEgressCIDR{},
 	}
 
+	// node-local-dns is not supported on openshift
 	if r.provider != operatorv1.ProviderOpenShift {
 		nodeLocalDNSExists, err := utils.IsNodeLocalDNSAvailable(ctx, r.client)
 		if err != nil {
 			r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying node-local-dns pods", err, reqLogger)
 			return nil, &reconcile.Result{RequeueAfter: 10 * time.Second}
 		} else if nodeLocalDNSExists {
-			// Discover the kube-dns Service cluster IP address - node-local-dns is not supported on OpenShift which is the only platform without
-			// kube-dns. Thus, the name "kube-dns" can be static.
-			kubeDNSService := &corev1.Service{}
-			err = r.client.Get(ctx, types.NamespacedName{Name: "kube-dns", Namespace: "kube-system"}, kubeDNSService)
+			dnsServiceIPs, err := utils.GetDNSServiceIPs(ctx, r.client, r.provider)
 			if err != nil {
-				if errors.IsNotFound(err) {
-					r.status.SetDegraded(operatorv1.ResourceNotFound, "kube-dns service not found", err, reqLogger)
+				if apierrors.IsNotFound(err) {
+					r.status.SetDegraded(operatorv1.ResourceNotFound, "DNS service is not found", err, reqLogger)
```

Comment:
Just a nit, but "DNS service is not found" isn't quite right syntax-wise.

I'd pick one of these:

- DNS service cannot be found
- Unable to find DNS service
- DNS service not found